From 05fdcf5ae840b52ce94b8c2aebfe49c30e2e4388 Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Tue, 2 Mar 2021 23:14:16 +0530
Subject: [PATCH 1282/1921] crypto: marvell: re-enable MBOX interrupt in FLR
 handler

Re-enable MBOX interrupt in FLR handler as it gets cleared
in HWVF_RST reset,
commit 9fb681694c1f ("octeontx2-af: Reset MSIX config space in FLR")

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: I8c1a3afff7b3e8212fc9237f4ece84715568bc97
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47018
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c    | 4 ++++
 drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c b/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c
index 9e31fb8b38d9..858864b5b22c 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c
+++ b/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c
@@ -135,6 +135,10 @@ static void cptpf_flr_wq_handler(struct work_struct *work)
 			  RVU_PF_VFTRPENDX(reg), BIT_ULL(vf));
 	cn10k_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
 			  RVU_PF_VFFLR_INT_ENA_W1SX(reg), BIT_ULL(vf));
+
+	/* Re-enable MBOX interrupt as it gets cleared in HWVF_RST reset */
+	cn10k_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
+			  RVU_PF_VFPF_MBOX_INT_ENA_W1SX(reg), BIT_ULL(vf));
 }
 
 static irqreturn_t cptpf_vf_flr_intr(int __always_unused irq, void *arg)
diff --git a/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c b/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c
index 6455f6bdc64b..669844aeda36 100644
--- a/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c
+++ b/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c
@@ -141,6 +141,10 @@ static void cptpf_flr_wq_handler(struct work_struct *work)
 			 RVU_PF_VFTRPENDX(reg), BIT_ULL(vf));
 	otx2_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
 			 RVU_PF_VFFLR_INT_ENA_W1SX(reg), BIT_ULL(vf));
+
+	/* Re-enable MBOX interrupt as it gets cleared in HWVF_RST reset */
+	otx2_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
+			 RVU_PF_VFPF_MBOX_INT_ENA_W1SX(reg), BIT_ULL(vf));
 }
 
 static irqreturn_t cptpf_vf_flr_intr(int __always_unused irq, void *arg)
-- 
2.31.1

